### ISSUE TITLE
T5 모델을 이용한 실험

### ISSUE THREAD

**Issue by kms7530** [boostcampaitech5/level3_nlp_finalproject-nlp-04#17]
# 👨‍🔬 실험 개요
- 위의 데이터셋을 기준으로 다음과 같은 설정으로 학습함. 
```
learning_rate=2e-5,
per_device_train_batch_size=8,
per_device_eval_batch_size=8,
weight_decay=0.01,
predict_with_generate=True,
fp16=True
```
- 총 128,000 steps(3.56 epochs) 학습 시켰으며, 총 4 epochs를 실행하여 했으나, 더 이상 loss가 줄어들지 않아 학습 중단. 
- 다음의 표는 학습 결과이다. 

    Step | Training Loss | Validation Loss | Rouge1 | Rouge2 | Rougel | Rougelsum | Gen Len
    -- | -- | -- | -- | -- | -- | -- | --
    2000 | 0.321900 | 0.312891 | 0.097100 | 0.026200 | 0.097500 | 0.097700 | 18.994800
    4000 | 0.279300 | 0.293201 | 0.096800 | 0.025400 | 0.097000 | 0.097200 | 19.000000
    6000 | 0.271500 | 0.279073 | 0.099900 | 0.028000 | 0.100100 | 0.100300 | 19.000000
    ... | ... | ... | ... | ... | ... | ... | ... | 
    124000 | 0.204800 | 0.208545 | 0.105900 | 0.028400 | 0.105800 | 0.106200 | 19.000000
    126000 | 0.195700 | 0.209584 | 0.106300 | 0.028400 | 0.106000 | 0.106900 | 19.000000
    128000 | 0.190200 | 0.206511 | 0.107700 | 0.028400 | 0.107400 | 0.107900 | 19.000000

# 📊 데이터셋 관점
- 학습 시 사용된 데이터는 다음과 같다. 
```
전문
2012년 경상남도 진주시 농산물도매시장에 판매중인 감귤에서 분홍빛열매썩음병이 발생하였다. 병징은 감귤 과실 표면이 수침상으로 물러지고 썩으면서 그 위에 분홍빛 곰팡이가 많이 형성되었다. 균총의 색깔은 처음에 흰색이고 배양기간이 경과됨에 따라 배지 표면에 분홍빛의분생포자가 많이 형성되었다. 균사생육 적온은 25oC이었다. 분생포자의 모양은 서양배형이며 좌우 zigzag로 부착하며 성숙한 분생포자는 2세포로 되어 있으며 크기는 12?26 × 8?12 μm이었다. 분생자경은 균사표면에 직립으로 형성하고, 폭이 4-5 μm이고 무색이었다. rDNA의 completeinternal transcribed spacer(ITS) 영역의 염기서열을 분석하였고, 분석된 염기서열(613 bp)을 BLASTN 프로그램으로확인한 결과, Trichothecium roseum와 99%의 상동성을 나타내었다. 이와 같이 감귤에서 발생한 병징과 병원균의균학적 특징을 기초로 하여 이 병을 Trichothecium roseum(Pers.) Link ex Gray에 의한 감귤 분홍빛열매썩음병으로명명하고자 제안한다.

요약
2012년 경상남도 진주시 농산물도매시장에 판매중인 감귤에서 분홍빛열매썩음병이 발생했다. 분생포자의 모양은 서양배형이며 좌우 zigzag로 부착하며 성숙한 분생포자는 2세포로 되어 있으며 크기는 12?26 × 8?12 μm이었다.
```

# 🚀 실제 적용
- 현재 학습된 모델을 이용하여 문장을 요약해본 결과 다음과 같다. 
- 먼저 원문이다. 
```
뚜렷한 이유는 안밝혀..지난해 사업협력 단순투자로 공시했는데네이버 카페 주주모임, 불매운동하자는 글까지3명 사외이사 재선임 최종 부결돼도상법상 임시주총까지 퇴임이사로 활동 가능[이데일리 김현아 기자]KT의 지분 7.79%(현대자동차 4.69%, 현대모비스 3.1%)를 보유한 현대자동차그룹이 오는 31일 KT 주주총회에서 KT 사외이사 3명의 재선임안(임기 1년)에 대해 반대하겠다는 뜻을 KT에 전달한 것으로 확인되자, ...
```

- 위의 기사는 Tokenizer로 tokenizing 한 경우 길이가 1200 이다. 이는 학습 시 사용된 원문의 최대 길이보다 긴 기사다. 
- 이를 요약하면 다음과 같다. 
```
뚜렷한 이유는 안밝혀..지난해 사업협력 단순투자로 공시했는데네이버 카페 주주모임, 불매운동하자는 글까지3명 사외이사 재선임 최종 부결돼도상법상 임시주총까지 퇴임이사로 활동 가능[이데일리 김현아 기자]KT의 지분 7.79%를 보유한 현대자동차그룹이 오는 31일 KT 주주총회에서 KT 사외이사 3명의 재선임안에 대해 반대하겠다는 뜻을 KT에 전달한
```

- 위의 요약된 예시를 보면 첫 문장을 그대로 가져온 후 문장이 끝나지 않은 체 생성을 멈추었다. 
- 이는 학습시 사용된 데이터셋이 이만큼 긴 문장에 대해 학습하지 않아, 길이가 넘어가는것에 대해 처리하지 못하는것으로 보인다. 
- 따라서 5개의 문장마다 나누어 실험 해 보았다. 

## 문장 단위 실험
- 뉴스 기사를 5개의 문장마다 나누어 모델에 요약을 시켰을 때 다음과 같은 결과로 나오게 된다. 
    - 문장 위의 각 숫자는 tokenizing 된 token의 길이다. 

```
436
[{'summary_text': 'KT의 지분 7.79%를 보유한 현대자동차그룹이 오는 31일 KT 주주총회에서 KT 사외이사 3명의 재선임안에 대해 반대하겠다는 뜻을 KT에 전달한 것으로 확인되자, KT 개인주주들이 분노하고 있다. 현대차그룹은 강충구 고려대 전기전자공학부 교수, 여은정 중앙대 경영학부 교수, 표현명 한국타이어테크놀로지 사외이사 등 3명에 대해                                                                                        '}]
254
[{'summary_text': '네이버 KT주주모임 카페에서 아이디 알바트로스님은 “현대차가 선을 넘는건가요?”라고 했고, chsu6366님은 “현대차그룹의 양아치 경영에 분노를 표하며 주주들이 불매운동에 적극적으로 동참하시길 당부드린다. KT CFO는 현대차그룹 지분을 매각하시길 바란다”고 했다. 앞서 주주모임 카페 개설자도 이데일리와의 통화에서 �'}]
265
[{'summary_text': '강충구, 여은정, 표현명 이사가 주총에서 재선임되지 않으면 이사는 1명만 남게 된다. 현대차 반대해 최종 부결돼도 상법으로 이사 의무 유지하지만, 법조계에 따르면 설사 KT 이사가 한 명도 안 남아도 상법상 이사의 결원 조항으로 인해 새 이사회 구성까지 임무를 담당할 수 있다. 퇴임이사 자격으로 가능하다.\n\n퇴임이사 자격으로 '}]
252
[{'summary_text': 'KT 이사회의 최소 의결 정족 원수는 3명인데, 현대차 주장대로 선임안이 부결되면 이사는 1명 남게 된다. 하지만, 부결돼도 해당 이사들은 이 조항에 근거해 임시주총에서 새로운 이사가 추천돼 선임될 때까지 퇴임이사로 활동할 수 있다. 한편 KT의 주가는 다시 3만 원대 아래로 추락했다. 29일 현재 2만 9200원이 '}]
```

- 위의 실험 또한 문장이 아직 끝나지 않았는데, 완결된 문장만 가져오게 되면 다음과 같다. 
```
KT의 지분 7.79%를 보유한 현대자동차그룹이 오는 31일 KT 주주총회에서 KT 사외이사 3명의 재선임안에 대해 반대하겠다는 뜻을 KT에 전달한 것으로 확인되자, KT 개인주주들이 분노하고 있다.

네이버 KT주주모임 카페에서 아이디 알바트로스님은 “현대차가 선을 넘는건가요?”라고 했고, chsu6366님은 “현대차그룹의 양아치 경영에 분노를 표하며 주주들이 불매운동에 적극적으로 동참하시길 당부드린다. KT CFO는 현대차그룹 지분을 매각하시길 바란다”고 했다.

강충구, 여은정, 표현명 이사가 주총에서 재선임되지 않으면 이사는 1명만 남게 된다. 현대차 반대해 최종 부결돼도 상법으로 이사 의무 유지하지만, 법조계에 따르면 설사 KT 이사가 한 명도 안 남아도 상법상 이사의 결원 조항으로 인해 새 이사회 구성까지 임무를 담당할 수 있다. 퇴임이사 자격으로 가능하다.

KT 이사회의 최소 의결 정족 원수는 3명인데, 현대차 주장대로 선임안이 부결되면 이사는 1명 남게 된다. 하지만, 부결돼도 해당 이사들은 이 조항에 근거해 임시주총에서 새로운 이사가 추천돼 선임될 때까지 퇴임이사로 활동할 수 있다. 한편 KT의 주가는 다시 3만 원대 아래로 추락했다.
```

- 이를 이전의 요약문과 비교해 보았을 때 더 나은 요약 결과를 주었다. 
    - 결과로 보았을 때 추후 실험을 통해 원문 자체를 주었을 때 좋은 성능을 보여주지 못하는 경우 해당 방법을 사용하면 될 것으로 보인다.
